compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -806,10 +806,11 @@ trait Checking {
       val problem = if (pat.tpe <:< reportedPt) "is more specialized than" else "does not match"
       val fix = if (isPatDef) "adding `: @unchecked` after the expression" else "writing `case ` before the full pattern"
       val pos = if (isPatDef) sel.srcPos else pat.srcPos
+      def rewriteMsg = Message.rewriteNotice("This patch", `future-migration`)
       report.warning(
         ex"""pattern's type ${pat.tpe} $problem the right hand side expression's type $reportedPt
             |
-            |If the narrowing is intentional, this can be communicated by $fix.${err.rewriteNotice}""",
+            |If the narrowing is intentional, this can be communicated by $fix.$rewriteMsg""",
```

Comment:
Sorry my comment will be out of scope, but as a feature request, I wonder if there is a syntax to make string construction more intuitive or easier to read. Just thinking out loud, what if the rewrite message were `given` so that it needn't be stuck at the end by hand.